windows/test.bash

```diff
@@ -0,0 +1,35 @@
+#!/usr/bin/env bash
+
+set -eo pipefail
+
+PROTOC_VERSION="3.19.1"
+PROTOC_GEN_GO_VERSION="v1.27.1"
+PROTOC_GEN_GO_GRPC_VERSION="30dfb4b933a50fd366d7ed36ed4f71dbba2d382e"
+
+# Convert DOWNLOAD_CACHE from d:\path to /d/path
+DOWNLOAD_CACHE="$(echo "/${DOWNLOAD_CACHE}" | sed 's|\\|/|g' | sed 's/://')"
+mkdir -p "${DOWNLOAD_CACHE}"
+PATH="${DOWNLOAD_CACHE}/protoc/bin:${PATH}"
+
+if [ -f "${DOWNLOAD_CACHE}/protoc/bin/protoc.exe" ]; then
+  CACHED_PROTOC_VERSION="$("${DOWNLOAD_CACHE}/protoc/bin/protoc.exe" --version | cut -d " " -f 2)"
+fi
+
+if [ "${CACHED_PROTOC_VERSION}" != "$PROTOC_VERSION" ]; then
+  PROTOC_URL="https://github.com/protocolbuffers/protobuf/releases/download/v${PROTOC_VERSION}/protoc-${PROTOC_VERSION}-win64.zip"
+  curl -sSL -o "${DOWNLOAD_CACHE}/protoc.zip" "${PROTOC_URL}"
+  7z x -y -o"${DOWNLOAD_CACHE}/protoc" "${DOWNLOAD_CACHE}/protoc.zip"
+  mkdir -p "${DOWNLOAD_CACHE}/protoc/lib"
+  cp -a "${DOWNLOAD_CACHE}/protoc/include" "${DOWNLOAD_CACHE}/protoc/lib/include"
+else
+  echo "Using cached protoc"
+fi
+
+PATH="${DOWNLOAD_CACHE}/protoc/bin:${PATH}"
+
+go install google.golang.org/protobuf/cmd/protoc-gen-go@${PROTOC_GEN_GO_VERSION}
+go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@${PROTOC_GEN_GO_GRPC_VERSION}
+go install ./private/buf/cmd/buf/command/protoc/internal/protoc-gen-insertion-point-writer
+go install ./private/buf/cmd/buf/command/protoc/internal/protoc-gen-insertion-point-receiver
+go install ./cmd/buf
```

Comment:
Should this be one `go install` command with each parameter separated by `\` on newlines?

Reply:
Thanks, I didn't know `go install` worked with multiple packages.

I combined the three local packages into one command, but that trick apparently doesn't work when you specify different versions for the modules.

```shell
$ go install google.golang.org/protobuf/cmd/protoc-gen-go@v1.27.1 \
  google.golang.org/grpc/cmd/protoc-gen-go-grpc@30dfb4b933a50fd366d7ed36ed4f71dbba2d382e
go install: google.golang.org/grpc/cmd/protoc-gen-go-grpc@30dfb4b933a50fd366d7ed36ed4f71dbba2d382e: all arguments must have the same version (@v1.27.1)
```

Reply:
Ah got it, thanks for sharing that.